src/components/swap/SwapRoute.tsx

```diff
@@ -1,47 +1,75 @@
+import { RowBetween } from 'components/Row'
+import { Version } from 'hooks/useToggledVersion'
+import { memo } from 'react'
+import styled from 'styled-components/macro'
+import { TYPE } from 'theme'
+import { getTradeVersion } from 'utils/getTradeVersion'
+import { RouterLabel } from './RouterLabel'
 
-function LabeledArrow({}: { fee: FeeAmount }) {
-  const theme = useContext(ThemeContext)
-
-  // todo: render the fee in the label
-  return <ChevronRight size={14} color={theme.text2} />
-}
+const Separator = styled.div`
+  border-top: 1px solid ${({ theme }) => theme.bg2};
+  width: 100%;
+  height: 1px;
+`
 
 export default memo(function SwapRoute({
   trade,
 }: {
   trade: V2Trade<Currency, Currency, TradeType> | V3Trade<Currency, Currency, TradeType>
 }) {
-  const tokenPath = trade instanceof V2Trade ? trade.route.path : trade.route.tokenPath
-  const theme = useContext(ThemeContext)
   return (
-    <Flex flexWrap="wrap" width="100%" justifyContent="flex-start" alignItems="center">
-      {tokenPath.map((token, i, path) => {
-        const isLastItem: boolean = i === path.length - 1
-        const currency = unwrappedToken(token)
-        return (
-          <Fragment key={i}>
-            <Flex alignItems="end">
-              <TYPE.black color={theme.text1} ml="0.145rem" mr="0.145rem">
-                {currency.symbol}
-              </TYPE.black>
-            </Flex>
-            {isLastItem ? null : trade instanceof V2Trade ? (
-              <ChevronRight size={14} color={theme.text2} />
-            ) : (
-              <LabeledArrow fee={trade.route.pools[i].fee} />
-            )}
-          </Fragment>
-        )
-      })}
-    </Flex>
+    <AutoColumn gap="12px" style={{ padding: '0.5rem 0' }}>
+      <RowBetween>
+        <RouterLabel />
+        <TYPE.black fontSize={14}>
+          {getTradeVersion(trade) === Version.v2 ? <Trans>Best route via V2</Trans> : <Trans>Best route via V3</Trans>}
+        </TYPE.black>
+      </RowBetween>
+      <Separator />
+      <RoutingDiagram
+        currencyIn={trade.inputAmount.currency}
+        currencyOut={trade.outputAmount.currency}
+        routes={getTokenPath(trade)}
+      />
+    </AutoColumn>
   )
 })
+
+function getTokenPath(
+  trade: V2Trade<Currency, Currency, TradeType> | V3Trade<Currency, Currency, TradeType>
+): RoutingDiagramEntry[] {
+  // convert V2 path to a list of routes
+  if (trade instanceof V2Trade) {
+    const { path: tokenPath } = (trade as V2Trade<Currency, Currency, TradeType>).route
+    const path = []
+    for (let i = 1; i < tokenPath.length; i++) {
+      path.push([tokenPath[i - 1], tokenPath[i], undefined] as [Currency, Currency, FeeAmount | undefined])
+    }
+    return [{ percent: new Percent(100, 100), path }]
+  }
+
+  return trade.swaps.map(({ route: { tokenPath, pools }, inputAmount }) => {
+    const portion = inputAmount.divide(trade.inputAmount)
+    const percent = new Percent(portion.numerator, portion.denominator)
```

Comment:
why not just `new Percent(inputAmount, trade.inputAmount)`?

Reply:
hm i'd need to format `inputAmount` into a `BigIntIsh`. Creating the percent from the fraction seemed more straightfoward.